lib/ansible/modules/crypto/get_certificate.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: get_certificate
+author: "John Westcott IV"
+version_added: "2.7"
+short_description: Get a certificate from a host:port
+description:
+    - Makes an SSL connection and returns information about the presented certificate
+options:
+    host:
+      description:
+        - The host to get the cert for (IP is fine)
+      required: True
+    ca_certs:
+      description:
+        - A file containing a list of root certificates; if present, the cert will be validated against these root certs.
+      required: False
+    port:
+      description:
+        - The port to connect to
+      required: True
+    timeout:
+      description:
+        - The timeout in seconds
+      required: False
+      default: 10
+
+notes:
+  - When using ca_certs on OS X it has been reported that in some conditions the validate will always suceed.
+
+requirements:
+  - "python >= 2.6"
+  - "ssl"
+  - "python-pyOpenSSL >= 0.15"
+  - "socket"
+'''
+
+RETURN = '''
+cert:
+    description: Information about the cert (CN, OU, etc)
+    returned: success
+    type: dict
+expired:
+    description: Boolean indicating if the cert is expired
+    returned: success
+    type: bool
+extensions:
+    description: Extensions applied to the cert
+    returned: success
+    type: list
+issuer:
+    description: Information about the issuer of the cert
+    returned: success
+    type: dict
+not_after:
+    description: Expiration date of the cert
+    returned: success
+    type: string
+not_before:
+    description: Issue date of the cert
+    returned: success
+    type: string
+serial_number:
+    description: The searial number of the cert
+    returned: success
+    type: string
+signature_algorithm:
+    description: The algorithm used to sign the cert
+    returned: success
+    type: string
+subject:
+    description: An alias of 'cert'
+    returned: success
+    type: dict
+version:
+    description: The version number of the certificate
+    returned: success
+    type: string
+'''
+
+EXAMPLES = '''
+- name: Get the cert from an RDP port
+  get_certificate:
+    host: "1.2.3.4"
+    port: 3389
+  delegate_to: localhost
+  run_once: true
```

Comment:
Why the `run_once`?

Reply:
Because I have delegate_to set to localhost. If you were running on three target nodes with out run_once you will run this step 3 times.